src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -655,41 +657,35 @@ def _visit_iteration_order_node(self, node: ast.withitem, loc: gt_ir.Location):
         return gt_ir.IterationOrder[iteration_order_node.id]
 
     def _visit_interval_node(self, node: ast.withitem, loc: gt_ir.Location):
-        # initialize possible exceptions
-        interval_error = GTScriptSyntaxError(
-            f"Invalid 'interval' specification at line {loc.line} (column {loc.column})", loc=loc
-        )
         range_error = GTScriptSyntaxError(
             f"Invalid interval range specification at line {loc.line} (column {loc.column})",
             loc=loc,
         )
 
-        interval_node = node.context_expr
+        if node.context_expr.args:
+            args = node.context_expr.args
+        else:
+            args = [keyword.value for keyword in node.context_expr.keywords]
+            if len(args) != 2:
+                raise range_error
+
+        if len(args) == 2:
+            interval_node = ast.Slice(lower=args[0], upper=args[1])
+            ast.copy_location(interval_node, node)
+        else:
+            interval_node = args[0]
+
+        interval = gt_ir.utils.parse_interval_node(interval_node, "K", loc=loc)
 
-        # validate interval specification
         if (
-            (len(interval_node.args) + len(interval_node.keywords) < 1)
-            or (len(interval_node.args) + len(interval_node.keywords) > 2)
-            or any(keyword.arg not in ["start", "end"] for keyword in interval_node.keywords)
+            interval.start.level == gt_ir.LevelMarker.END
+            and interval.end.level == gt_ir.LevelMarker.START
         ):
-            raise interval_error
-
-        # parse interval
-        if interval_node.args:
-            range_node = interval_node.args
-        else:
-            range_node = [interval_node.keywords[0].value, interval_node.keywords[1].value]
-        if len(range_node) == 1 and isinstance(range_node[0], ast.Ellipsis):
-            interval = gt_ir.AxisInterval.full_interval()
-        elif len(range_node) == 2 and all(
-            isinstance(elem, (ast.Num, ast.UnaryOp, ast.NameConstant)) for elem in range_node
+            raise range_error
+        elif (
+            interval.start.level == interval.end.level
+            and interval.end.offset <= interval.start.offset
```

Comment:
I think it would be cleaner to  merge these two nested ifs in a single one using `or`